### ISSUE TITLE
[CLI/SDK 49 Beta] Windows PowerShell issue when pressing `w` to open browser

### ISSUE THREAD

**Issue by danilopolani** [expo/expo#23252]
### Summary

I was trying the SDK 49 Beta 3, but on Windows it seems that something does not work with the CLI.  
First of all: running `yarn start` produces this error:

> /bin/bash: D:\<myproject>\node_modules\.bin\..\expo\bin\cli: No such file or directory
error Command failed with exit code 127.

After investigating, I noticed that from SDK 48 the CLI changed and now invokes a `cli` bash file that proxies all the call to `expo internal`, so I decided to run that command directly:

> npm exec --no-install -- expo-internal "start"  

It works! But, pressing `w` to open Web, it throws this error:

> Error: spawn undefined\System32\WindowsPowerShell\v1.0\powershell ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

Note that I've tried with both Cmder and PowerShell and until SDK 48 it just worked fine.  
The only way to make it "work" is to run commands inside `Git bash` shell. I've put `work` inside quotes because it just starts the Web process, without showing all the usual menu with QRCode etc., don't know why.

### What platform(s) does this occur on?

Web

### SDK Version

49 beta 3

### Environment

System:
      OS: Windows 10 10.0.22621
    Binaries:
      Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
      npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~49.0.0-beta.3 => 49.0.0-beta.3
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.72.0 => 0.72.0
      react-native-web: ~0.19.6 => 0.19.6
    Expo Workflow: managed

### Minimal reproducible example

Unable to provide it, sorry!

**Comment by byCedric** [comment]
Might be fixed by https://github.com/expo/expo/pull/23220

**Comment by danilopolani** [comment]
> Might be (partially) fixed by #23220

I've tried that patch by combining it with editing `expo.cmd` like this:

```
@IF EXIST "%~dp0\node.exe" (
  "%~dp0\node.exe"  "%~dp0\..\expo\bin\cli" %*
) ELSE (
  @SETLOCAL
  @SET PATHEXT=%PATHEXT:;.JS;=;%
  node  "%~dp0\..\expo\bin\cli" %*
)
```

Now the Expo process starts correctly, but still gives error when spawning PowerShell. It's still something tho :)

**Comment by byCedric** [comment]
Awesome! Thanks for testing that fix @danilopolani. I'll test opening Web with PowerShell a bit more on Windows. Do you know what version of PowerShell you are using by any chance?

**Comment by danilopolani** [comment]
I'm using PowerShell 5.1.22621.1778, but I don't know if the issue is related to PowerShell itself or, probably, to the spawn of the process from Node.

What's weird to me is that `undefined` in front of the path, also I can't find any occurrence of those flags sent along with the command in the `expo/sdk49` branch, maybe they're coming from an external package, idk

```
Error: spawn undefined\System32\WindowsPowerShell\v1.0\powershell ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn undefined\\System32\\WindowsPowerShell\\v1.0\\powershell',
  path: 'undefined\\System32\\WindowsPowerShell\\v1.0\\powershell',
  spawnargs: [
    '-NoProfile',
    '-NonInteractive',
    '–ExecutionPolicy',
    'Bypass',
    '-EncodedCommand',
    'something here'
  ]

**Comment by byCedric** [comment]
Fix #23220 is released in `expo@49.0.0-beta.4`

**Comment by byCedric** [comment]
@brentvatne the issue originally included the error related to "expo needs to be installed", which is fixed by #23220. But, this issue now focuses on opening browser issues on Windows.

I found the issue, will open a PR soon.

**Comment by byCedric** [comment]
@danilopolani this should be resolved in `expo@49.0.0` 😄

**Comment by danilopolani** [comment]
> @danilopolani this should be resolved in `expo@49.0.0` 😄 Thanks for reporting it and helping test the fixes.

Huh stable release? Nice! It's solved now, thank you for the quick fix and effort :)
